### ISSUE TITLE
Should localReceiptData be updated when a Subscription auto renews?

### ISSUE THREAD

**Issue by chamitha** [bizz84/SwiftyStoreKit#603]
## Question
I am testing auto-renewing subscriptions in the Sandbox and have noticed that my `SwiftyStoreKit.completeTransactions` completion handler in `AppDelegate` is invoked with a `purchased` state as expected when a subscription auto renews BUT subsequently the `SwiftyStoreKit.localReceiptData` property seems to not have been updated as `SwiftyStoreKit.verifySubscription` returns a status of expired with the old subscription expiry date. Is this expected?
Of course if I were to refresh the receipt (using either fetchReceipt or verifyReceipt with forceRefresh) the latest (renewed and active) subscription is received.

**Comment by chamitha** [comment]
> Hey, have you managed to find a way there? Was this resolved? Is simply calling fetchReceipt each time this happens a solution?

@Reiszecke Yes, I did end up calling `fetchReceipt(forceRefresh: true)`

**Comment by chamitha** [comment]
> @chamitha thank you for responding. If you can take 3 more minutes of your spare time:
> 
@Reiszecke 
> * do you get prompted an AppleID-signin when doing that?
Yes, `fetchReceipt(forceRefresh: true)` will prompt for sign-in if not already done so. But I don't think`SwiftyStoreKit.completeTransactions` is called when not signed-in? In which case no one will be prompted.

> * merchantkit takes into account 30 days of App Store delay with the receipts - do you also experience that issue?
I haven't used merchantkit. 

> * my users suffer from [Restore outdated subscription and then Purchase, SwiftyStoreKit refuse to proceed #587](https://github.com/bizz84/SwiftyStoreKit/issues/587) in production; is that also an issue in your app?
I can't say for sure. I have since migrated to manage in app purchases server side so these issues have dropped off my radar. Sorry I can't be of more help 😞

**Comment by kodboy** [comment]
If you start a development demo from another directory (such as the download directory), its receipt will not be updated. This is because of Apple's security mechanism. Apps in other directories will be copied to a private directory before starting. Apple updates the latest receipt to the receipt file in the app in the download directory, but when reading the receipt, it reads the receipt file of the app in the private directory. The solution is to copy the demo to the Applications directory and then start it.
